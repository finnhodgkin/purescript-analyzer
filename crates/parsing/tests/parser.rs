macro_rules! parser_tests {
    ($($name:ident => $source:expr),+ $(,)?) => {
        $(
            #[test]
            fn $name() {
                let lexed = lexing::lex($source);
                let tokens = lexing::layout(&lexed);
                let (node, errors) = parsing::parse(&lexed, &tokens);
                insta::assert_debug_snapshot!((node, errors));
            }
        )+
    };
}

parser_tests!(
    comment_pre_post => include_str!("parser/CommentPrePost.purs"),
    lexer_error_inserted => include_str!("parser/LexerErrorInserted.purs"),
    module_export => include_str!("parser/ModuleExport.purs"),
    module_export_class => include_str!("parser/ModuleExportClass.purs"),
    module_export_class_error_end => include_str!("parser/ModuleExportClassErrorEnd.purs"),
    module_export_comma_error => include_str!("parser/ModuleExportCommaError.purs"),
    module_export_data => include_str!("parser/ModuleExportData.purs"),
    module_export_data_comma_error => include_str!("parser/ModuleExportDataCommaError.purs"),
    module_export_data_item_after_double_period => include_str!("parser/ModuleExportDataItemAfterDoublePeriod.purs"),
    module_export_data_item_before_double_period => include_str!("parser/ModuleExportDataItemBeforeDoublePeriod.purs"),
    module_export_data_item_invalid => include_str!("parser/ModuleExportDataItemInvalid.purs"),
    module_export_empty => include_str!("parser/ModuleExportEmpty.purs"),
    module_export_invalid => include_str!("parser/ModuleExportInvalid.purs"),
    module_export_type_operator => include_str!("parser/ModuleExportTypeOperator.purs"),
    module_export_type_operator_error => include_str!("parser/ModuleExportTypeOperatorError.purs"),
    module_export_type_operator_error_end => include_str!("parser/ModuleExportTypeOperatorErrorEnd.purs"),
    module_header => include_str!("parser/ModuleHeader.purs"),
    module_header_prefixed => include_str!("parser/ModuleHeaderPrefixed.purs"),
);
