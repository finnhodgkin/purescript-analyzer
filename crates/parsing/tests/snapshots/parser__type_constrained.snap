---
source: crates/parsing/tests/parser.rs
expression: "(node, errors)"
snapshot_kind: text
---
(
    Module@0..79
      ModuleHeader@0..30
        MODULE@0..6 "module"
        Comment@6..7
          WHITESPACE@6..7 " "
        ModuleName@7..23
          UPPER@7..22 "TypeConstrained"
          Comment@22..23
            WHITESPACE@22..23 " "
        ModuleExportList@23..23
        WHERE@23..28 "where"
        Comment@28..30
          WHITESPACE@28..30 "\n\n"
      LAYOUT_START@30..30 ""
      ModuleImports@30..30
      ModuleStatements@30..79
        ValueAnnotation@30..52
          LOWER@30..41 "constrained"
          Comment@41..42
            WHITESPACE@41..42 " "
          DOUBLE_COLON@42..44 "::"
          Comment@44..45
            WHITESPACE@44..45 " "
          TypeConstrained@45..52
            TypeConstructor@45..47
              QualifiedName@45..47
                UPPER@45..46 "A"
                Comment@46..47
                  WHITESPACE@46..47 " "
            RIGHT_THICK_ARROW@47..49 "=>"
            Comment@49..50
              WHITESPACE@49..50 " "
            TypeConstructor@50..52
              QualifiedName@50..52
                UPPER@50..51 "B"
                Comment@51..52
                  WHITESPACE@51..52 "\n"
        LAYOUT_SEPARATOR@52..52 ""
        ValueAnnotation@52..79
          LOWER@52..63 "constrained"
          Comment@63..64
            WHITESPACE@63..64 " "
          DOUBLE_COLON@64..66 "::"
          Comment@66..67
            WHITESPACE@66..67 " "
          TypeConstrained@67..79
            TypeConstructor@67..69
              QualifiedName@67..69
                UPPER@67..68 "A"
                Comment@68..69
                  WHITESPACE@68..69 " "
            RIGHT_THICK_ARROW@69..71 "=>"
            Comment@71..72
              WHITESPACE@71..72 " "
            TypeConstrained@72..79
              TypeConstructor@72..74
                QualifiedName@72..74
                  UPPER@72..73 "B"
                  Comment@73..74
                    WHITESPACE@73..74 " "
              RIGHT_THICK_ARROW@74..76 "=>"
              Comment@76..77
                WHITESPACE@76..77 " "
              TypeConstructor@77..79
                QualifiedName@77..79
                  UPPER@77..78 "C"
                  Comment@78..79
                    WHITESPACE@78..79 "\n"
      LAYOUT_END@79..79 ""
      END_OF_FILE@79..79 ""
    ,
    [],
)
